src/core/sig_state.ml

```diff
@@ -160,14 +157,15 @@ let update_pp_hints_from_symbols :
     let h =
       let exception Hint of pp_hint in
       try
-        let fn_binop _ (s,binop) =
-          if s == sym then raise (Hint (Infix binop))
+        let fn s synt =
+          match synt with
+          | Sign.Infix binop ->
+              if s == sym then raise (Hint (Infix binop))
+          | Sign.Prefix unop ->
+              if s == sym then raise (Hint (Prefix unop))
+          | _ -> ()
         in
-        StrMap.iter fn_binop !(sign.sign_binops);
-        let fn_unop _ (s,unop) =
-          if s == sym then raise (Hint (Prefix unop))
-        in
-        StrMap.iter fn_unop !(sign.sign_unops);
+        SymMap.iter fn !(sign.sign_syntax);
```

Comment:
It seems to me that we can simplify this function a lot now that we have sign_syntax and pp_hints are attached to symbols instead of strings. In fact, it can be reduced (and thus unfolded) to:
```
SymMap.fold SymMap.add sign.sign_syntax pp_hints
```
If you take sign_syntax:pp_hint SymMap.t.